tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageChecks.cs

```diff
@@ -366,6 +367,30 @@ protected async Task<CLICheckResponse> CheckSpellingCommonAsync(string packagePa
 
     public virtual string GetSDKPackagePath(string repo, string packagePath)
     {
+        // For JavaScript packages, read the package name from package.json
+        var packageJsonPath = Path.Combine(packagePath, "package.json");
+        if (File.Exists(packageJsonPath))
```

Comment:
if we want to avoid checking the language this way we could also make this (GetSDKPackagePath) a method in ILanguageSpecificChecks w/ a default implementation